src/test/java/space/obminyashka/items_exchange/end2end/UserFlowTest.java

```diff
@@ -141,19 +141,31 @@ void getFavoriteAdvertisements_shouldReturnPage_WhenHaveSpecialPageAndSize() thr
                 .andExpect(jsonPath("$.content[0].title").value("Blouses"));
     }
 
+    @Test
+    @WithMockUser(username = ADMIN_USERNAME)
+    @DataSet("database_init.yml")
+    @ExpectedDataSet(value = "advertisement/addFavorite.yml")
+    void addFavoriteAdvertisement_shouldReturnException_whenAdvertisementIsNotFound() throws Exception {
```

Comment:
Do we have any positive end2end test for that logic?

Reply:
+